AI Generated Quizzes from Material URL

holoisme opened the pull request:
Hello,

This PR adds the ability to generate quizzes from material PDF documents.
The scope of this PR is only concerned with the back-end, so everything from PDF parsing, proper task prompt and raw AI answer parsing. The front-end by @FerdinandGomerChiu still needs to be able to receive it and will come in the following days.

FerdinandGomerChiu commented on edweiss-firebase/functions/src/functions/action/generateQuizContentFromMaterial.ts:
```diff
@@ -0,0 +1,138 @@
+		return fail("pdf-content-undefined");
+
+	/**
+questionmcq<TOKEN_END_QUESTION>
```
A nitpick: this comment can be removed.

holoisme replied:
I think it should instead be formally explicited as the "question" format for potential future refactors.

FerdinandGomerChiu replied:
Ah, I see! Thanks for clearing up the confusion. UwU

FerdinandGomerChiu commented:
I see you changed the toggleQuizResult cloud function, and added a new type in the lecture model. Have you made sure to not create conflicts with @adamm-alaoui-pro , as he is also working with those files?

holoisme commented:
> I see you changed the toggleQuizResult cloud function, and added a new type in the lecture model. Have you made sure to not create conflicts with @adamm-alaoui-pro , as he is also working with those files?

I didn't do much in those files, I just corrected his force-merge mistakes for me to compile my code.

FerdinandGomerChiu reviewed (approved):
This is insane work, especially considering you did it in about 30 min (after spending some time on flashcard generation). You are doing heavy lifting 🏋🏽. Keep it up!
